cfg/peer-filter/filter.go

```diff
@@ -0,0 +1,63 @@
+package filter
+
+import (
+	ds "github.com/jonhoo/drwmutex"
+
+	"github.com/libp2p/go-libp2p-circuit"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type PeerFilter struct {
```

Comment:
It would be nice to have a filter that checks network subnets as well, but that's good enough for now.

Reply:
Wrong comment sorry, its only used by utils, this allow the user to create filter where the In and Out filter are not from the same filter (example, if you want to create a filter filtering inbound with subnet and outbound with peer id, you can).